### PR TITLE
W2 - vietphan - Create calculator demo

### PR DESCRIPTION
**Pull request by at-vietphan**
#### Related issues (Optional)
Task #14 

#### Description (Required)
- [x] Create screen
- [x] handle event

### CODE REVIEW

**Comment by at-vietphan** [comment]
@at-huongnguyen Please review this pull request. Thanks :)

**Comment by at-vietphan** [comment]
@at-huongnguyen Please review this pull request. Thanks :)

File: app/src/at-vietphan/res/layout/activity_list_exercise.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        android:textColor="@android:color/black"
+        android:textSize="20dp"
+        android:textStyle="bold" />
+
+    <Button
+        android:id="@+id/btnCalculator"
+        style="@style/ButtonTheme"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="Calculator Demo" />
```
Comment (at-dev-support-01) [inline app/src/at-vietphan/res/layout/activity_list_exercise.xml]: WARNING: [HardcodedText] Hardcoded string "Calculator Demo", should use `@string` resource
 Hardcoding text attributes directly in layout files is bad for several reasons:

* When creating configuration variations (for example for landscape or portrait)you have to repeat the actual text (and keep it up to date when making changes)

* The application cannot be translated to other languages by just adding new translations for existing string resources.

There are quickfixes to automatically extract this hardcoded string into a resource lookup.

**Comment by at-vietphan** [comment]
@at-huongnguyen Please review this pull request. Thanks :)

File: app/src/at-vietphan/AndroidManifest.xml [hunk 2]
```diff
@@ -9,13 +9,14 @@
...
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity android:name=".CalculatorActivity"></activity>
     </application>
 
 </manifest>
```
Comment (at-huongnguyen) [inline app/src/at-vietphan/AndroidManifest.xml]: Add line break

**Comment by at-vietphan** [comment]
@at-huongnguyen Please review this pull request. Thanks :)

File: app/build.gradle [hunk 3]
```diff
@@ -79,6 +79,7 @@ android {
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation 'com.android.support:appcompat-v7:26.1.0'
+    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
```
Comment (at-huongnguyen) [inline app/build.gradle]: `constraint-layout` not use in this project, plz remove it

File: app/src/at-vietphan/java/vn/asiantech/internship/CalculatorActivity.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        mBtnMinus = findViewById(R.id.btnMinus);
+        mBtnMultiplication = findViewById(R.id.btnMultiplication);
+        mBtnDivision = findViewById(R.id.btnDivision);
+        mEdtNumberA = findViewById(R.id.edtNumberA);
+        mEdtNumberB = findViewById(R.id.edtNumberB);
+        mTvCalculation = findViewById(R.id.tvCalculation);
+        mTvResult = findViewById(R.id.tvResult);
+    }
+
+    private void implementClicks() {
```
Comment (at-huongnguyen) [inline app/src/at-vietphan/java/vn/asiantech/internship/CalculatorActivity.java]: Should rename to `initListener`

File: app/src/at-vietphan/java/vn/asiantech/internship/CalculatorActivity.java [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        mBtnMultiplication.setOnClickListener(this);
+        mBtnDivision.setOnClickListener(this);
+    }
+
+    private void setTextWatcher() {
+        mEdtNumberA.addTextChangedListener(textWatcher);
+        mEdtNumberB.addTextChangedListener(textWatcher);
+    }
+
+    private void isEnableBtn(Boolean aBoolean) {
```
Comment (at-huongnguyen) [inline app/src/at-vietphan/java/vn/asiantech/internship/CalculatorActivity.java]: Should rename to `setEnableButton(boolean isCheck)`

File: app/src/at-vietphan/java/vn/asiantech/internship/CalculatorActivity.java [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        }
+
+        @Override
+        public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+            // No-op
+        }
+
+        @Override
+        public void afterTextChanged(Editable editable) {
+            if (TextUtils.isEmpty(mEdtNumberA.getText()) || TextUtils.isEmpty(mEdtNumberB.getText())) {
```
Comment (at-huongnguyen) [inline app/src/at-vietphan/java/vn/asiantech/internship/CalculatorActivity.java]: Use in `onTextChanged ` method

File: app/src/at-vietphan/res/layout/activity_calculator.xml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    android:orientation="vertical"
+    android:paddingLeft="30dp"
+    android:paddingRight="20dp"
+    android:paddingTop="20dp"
+    tools:context="vn.asiantech.internship.CalculatorActivity">
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:gravity="fill_horizontal">
```
Comment (at-huongnguyen) [inline app/src/at-vietphan/res/layout/activity_calculator.xml]: This is redundancy

File: app/src/at-vietphan/res/layout/activity_calculator.xml [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+        <EditText
+            android:id="@+id/edtNumberA"
+            android:layout_width="match_parent"
+            android:layout_height="40dp"
+            android:layout_gravity="left"
```
Comment (at-huongnguyen) [inline app/src/at-vietphan/res/layout/activity_calculator.xml]: This is redundancy

File: app/src/at-vietphan/res/layout/activity_calculator.xml [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+            android:layout_width="46dp"
+            android:layout_height="match_parent"
+            android:gravity="center"
+            android:textSize="18dp" />
+
+        <EditText
+            android:id="@+id/edtNumberB"
+            android:layout_width="match_parent"
+            android:layout_height="40dp"
+            android:layout_gravity="right"
```
Comment (at-huongnguyen) [inline app/src/at-vietphan/res/layout/activity_calculator.xml]: This is redundancy

File: build.gradle [hunk 10]
```diff
@@ -7,7 +7,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.0'
+        classpath 'com.android.tools.build:gradle:3.0.1'
```
Comment (at-huongnguyen) [inline build.gradle]: Plz revert this

File: app/src/at-vietphan/res/values/styles.xml [hunk 11]
```diff
@@ -6,6 +6,12 @@
...
         <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
         <item name="colorAccent">@color/colorAccent</item>
-    </style>
+        <item name="android:buttonStyle">@style/ButtonTheme</item>
+        <item name="buttonStyle">@style/ButtonTheme</item>
 
+    </style>
+    <style name="ButtonTheme" parent="android:Widget.Button">
+        <item name="android:typeface">monospace</item>
+        <item name="android:textSize">16sp</item>
```
Comment (at-huongnguyen) [inline app/src/at-vietphan/res/values/styles.xml]: Use dp

File: app/src/at-vietphan/res/values/styles.xml [hunk 12]
```diff
@@ -6,6 +6,12 @@
         <item name="colorPrimary">@color/colorPrimary</item>
         <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
         <item name="colorAccent">@color/colorAccent</item>
-    </style>
+        <item name="android:buttonStyle">@style/ButtonTheme</item>
+        <item name="buttonStyle">@style/ButtonTheme</item>
 
+    </style>
+    <style name="ButtonTheme" parent="android:Widget.Button">
+        <item name="android:typeface">monospace</item>
```
Comment (at-huongnguyen) [inline app/src/at-vietphan/res/values/styles.xml]: Is not required

**Comment by at-vietphan** [comment]
@at-huongnguyen Please review this pull request. Thanks :+1:

**Comment by at-huongnguyen** [comment]
@at-vietphan 
Please resolve conflicts

**Pull request merged**